### DISCUSSION TITLE
Update user context after change browser tab

### DISCUSSION THREAD

**Discussion by HighError** [supabase/supabase#8758] [Questions] [answered]
I'm new in react and can't figure out why when I change the browser tab, the state is updated. I followed the example from the website https://ruanmartinelli.com/posts/supabase-authentication-react
What should be done to avoid this. As for the user, this is a simple update, but it will affect optimization in the future.
My code: https://github.com/HighError/Eviloma-Family-Dashboard

**Comment by GaryAustin1** [comment, marked as answer]
I assume you are talking about onAuthStateChange firing on every tab change.  That is the way it works to check for need to refresh token.
https://github.com/supabase/gotrue-js/issues/284

**Comment by HighError** [comment]
thx
